### PR TITLE
added documentation, SiteIntegrationAPI

### PR DESCRIPTION
**Pull request by kevinmram**
added documentation to SiteIntegrationAPI- documenting the purpose and usage of each method within the class.

### CODE REVIEW

File: backend/src/API/siteIntegrationAPI.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { PermissionError, BadRequestError } from '../utils/errors';
 
 require('dotenv').config();
 
+/**
+ * Triggers a dispatch event to pull changes from the IDOL backend.
```
Comment (henry-li-06) [inline backend/src/API/siteIntegrationAPI.ts:9]: I'd be a little more specific about what this does. 

e.g. Triggers a dispatch event for the pull-from-idol GitHub action. This job (blah blah blah brief description of what the job does).

File: backend/src/API/siteIntegrationAPI.ts [hunk 2]
```diff
@@ -19,12 +25,24 @@ export const requestIDOLPullDispatch = async (user: IdolMember): Promise<{ updat
...
+ * @throws {PermissionError} - If the user lacks permission to access the pull request.
+ */
 export const getIDOLChangesPR = async (user: IdolMember): Promise<{ pr: PRResponse }> => {
   await checkPermissions(user);
   const foundPR = await findBotPR();
   return { pr: foundPR };
 };
 
+/**
+ * Accepts and merges the IDOL backend changes pull request.
```
Comment (henry-li-06) [inline backend/src/API/siteIntegrationAPI.ts:41]: This should mention what happens when the PR is merged (i.e. triggering a deploy).

File: backend/src/API/siteIntegrationAPI.ts [hunk 3]
```diff
@@ -59,6 +77,12 @@ export const acceptIDOLChanges = async (
   return { pr: foundPR, merged: acceptedPR.data.merged };
 };
 
+/**
+ * Rejects and closes the IDOL backend changes pull request.
```
Comment (henry-li-06) [inline backend/src/API/siteIntegrationAPI.ts:81]:
```suggestion
 * Closes the IDOL backend changes pull request.
```

File: backend/src/API/siteIntegrationAPI.ts [hunk 4]
```diff
@@ -59,6 +77,12 @@ export const acceptIDOLChanges = async (
...
+ * @param {IdolMember} user - The user attempting to reject the changes.
+ * @returns {Promise<{pr: PRResponse; closed: boolean}>} - A promise that resolves to an object containing the pull request details and a boolean indicating if the rejection was successful.
+ * @throws {PermissionError} - If the user lacks permission to reject the pull request.
+ */
```
Comment (henry-li-06) [inline backend/src/API/siteIntegrationAPI.ts:85]: Also throws a BadRequestError if no PR is found.

**Comment by andrew032011** [comment]
@kevinmram doing some cleanup here to this PR doesn't get stale/abandoned. Let's either close this PR, or address these comments?

**Pull request merged**